### PR TITLE
added user authorization checks at service level

### PR DESCRIPTION
**Pull request by callunity**
for OICR sample hierarchy services

(Merge 1)

### CODE REVIEW

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/DefaultAuthorizationManager.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  @Override
+  public User getCurrentUser() throws IOException {
+    Authentication auth = securityContextHolderStrategy.getContext().getAuthentication();
+    if (auth == null) {
+      return null;
+    }
+    User user = securityManager.getUserByLoginName(auth.getName());
+    if (user == null && auth.isAuthenticated()) {
+      user = new UserImpl();
+      user.setAdmin(true);
```
Comment (callunity) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/DefaultAuthorizationManager.java]: Should there be an admin check as well before setting this?

**Comment by callunity** [comment]
:100: for authorizationManager!

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/DefaultAuthorizationManager.java (hunk 1 above)
Comment (djcooke) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/DefaultAuthorizationManager.java]: There's no way to do an admin check at this point. I'm not really sure of the case that this supports. The method was copied from UserAuthMisoRequestManager

Comment (callunity) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/security/DefaultAuthorizationManager.java]: Yeah, setting a null user to admin is bizarre. But if it's the pre-existing behaviour, I guess we go with that.

**Comment by callunity** [comment]
:+1:

**Comment by apmasell** [comment]
:+1: throws IOException

**Pull request merged**